source/common/protobuf/yaml_utility.cc

```diff
@@ -323,6 +323,28 @@ ProtobufWkt::Value ValueUtil::loadFromYaml(const std::string& yaml) {
   }
 }
 
+namespace {
+
+char* UTF8CoerceToStructurallyValid(absl::string_view str, char* dst, const char replace_char) {
+  // Copy the whole string (unless src == dst).
+  if (str.data() != dst) {
+    memmove(dst, str.data(), str.size());
+  }
+  char* const initial_dst = dst;
+  while (!str.empty()) {
+    int n_valid_bytes = static_cast<int>(utf8_range::SpanStructurallyValid(str));
+    if (n_valid_bytes == static_cast<int>(str.size())) {
+      break;
+    }
+    dst += n_valid_bytes;
+    *dst++ = replace_char; // replace one bad byte
+    str.remove_prefix(n_valid_bytes + 1);
+  }
+  return initial_dst;
+}
```

Comment:
Is this a copy of the function from the old protobuf version? Can we get a brief comment? It's a bit scary that `dst` here doesn't haven a length so we could potentially walk off the end of the buffer. It also looks like we're copying the buffer twice, once on line 364, then again on line 331. How 'bout if we change the signature to:

```
std::string UTF8CoerceToStructurallyValid(absl::string_view str, const char replace_char) {
```
This would create and return a new string which could ensure we don't overflow the string, and would only copy it once. WDYT?

Reply:
Sounds good